catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -155,18 +212,22 @@ private void fetchNextResults() {
             queryCopy.setStartIndex(currentIndex);
 
             try {
-                SourceResponse response = queryFunction.query(queryRequestCopy);
+                response = queryFunction.query(queryRequestCopy);
 
                 if (response.getResults()
                         .size() == 0) {
-                    results = RESULTS_EXHAUSTED;
+                    finished = true;
                     return;
                 }
 
                 results = response.getResults()
                         .iterator();
                 currentIndex += response.getResults()
                         .size();
+
```

Comment:
We can ensure `getHits() != -1 && currentIndex > response.getHits()` and if we're worried about it changing then we can store the first `getHits` we receive. Would that alleviate this issue?